src/main/java/com/networknt/schema/JsonMetaSchema.java

```diff
@@ -38,41 +38,47 @@
     private static final Logger logger = LoggerFactory
             .getLogger(JsonMetaSchema.class);
     private static Map<String, String> UNKNOWN_KEYWORDS = new ConcurrentHashMap<String, String>();
-    
-    private static class DraftV4 {
+
+    static PatternFormat pattern(String name, String regex) {
+        return new PatternFormat(name, regex);
+    }
+    public static final List<Format> BUILTIN_FORMATS = new ArrayList<Format>();
+
+    // this section contains formats that is common for all specification versions.
+    static {
+        BUILTIN_FORMATS.add(pattern("time", "^\\d{2}:\\d{2}:\\d{2}$"));

```

Comment:
It looks like we need to investigate the builtin formats a bit. The 'time' format isn't technically apart of draft 4 and there are other formats that we're not yet supporting such as 'duration'. So far the time-related formats are all backwards compatible since they follow RFC 3339 section 5.6. However, I'm worried that the formats aren't always backwards compatible, but I can't find any evidence of that yet. I wonder if it would be worth having different sets of builtin formats based on the chosen draft version.

draft 4 builtin formats:
https://tools.ietf.org/html/draft-fge-json-schema-validation-00#section-7.3 

draft 2019-09 builtin formats:
https://tools.ietf.org/html/draft-handrews-json-schema-validation-02#section-7.3

Reply:
I might have spoken too soon. Looks like you already started to allow draft specific formats. There is however a potential issue with the BUILTIN_FORMATS being shared by all of the V# classes. If the V4 and V6 classes both have their static blocks executed, then the BUILTIN_FORMATS will become polluted with both draft specific formats. That's a corner case that probably won't be encountered, but maybe someone would want to do it for whatever reason.

An easy remedy would be to have a static 'BUILTIN_FORMATS' variable in each of the V# classes and to utilize the formats from the previous draft version. This allows a single application to utilize JsonSchemas for different draft versions without having unexpected behavior. Our testing environment could have probably run into this problem since we're going to theoretically have to test all of the various draft versions.

Following is a quick snippet of what I was talking about. I assumed that builtin formats are backwards compatible, but I'm not entirely sure about that.
```
class JsonMetaSchema {
    public static final List<Format> BUILTIN_FORMATS = new ArrayList<>();
    static { ... }

    class V4 {
        public static final List<Format> BUILTIN_FORMATS = new ArrayList<>(JsonMetaSchema.BUILTIN_FORMATS);
        static { ... }
    }

    class V6 {
        public static final List<Format> BUILTIN_FORMATS = new ArrayList<>(JsonMetaSchema.V4.BUILTIN_FORMATS);
        static { ... }
    }
}
```

Reply:
Another thing, would it be a good idea to make an immutable list for the BUILTIN_FORMATS? I'm not sure if people do it, but it's technically possible for users to add to the BUILTIN_FORMATS. That might be a bad thing because it allows users to shoot themselves in the foot on accident. (Java 9's List.of() creates an immutable list if you want to use that.)

Reply:
You are right. I thought there is only one instance of the JsonMetaSchema that needs to be created, but we cannot guarantee that users won't create another instance. I will make the change.

Reply:
I think we want to BUILDIN_FORMATS to be open so that customized formats can be injected easily. I haven't thought about how this can be done yet.

Reply:
The JsonMetaSchema.BUILTIN_FORMATS is public and static, so any user can very easily inject custom formats. All it takes is something like:

```
class JsonSchemaFactoryFactory {
  public static JsonSchemaFactory create() {
    JsonMetaSchema.BUILTIN_FORMATS.add(new PatternFormat("new-keyword", "\\d\\d\\d"));
    // or
    JsonMetaSchema.BUILTIN_FORMATS.V7.add(new PatternFormat("new-keyword", "\\d\\d\\d"));

    // The BUILTIN_FORMATS needs to be updated prior to the creation of the JsonSchemaFactory. JsonMetaSchema will utilize a completely new formats list after it is created with the JsonMetaSchema.Builder (which is great and I like it that way).
    return JsonSchemaFactory.getInstance(SpecVersion.VersionFlag.V7);
  }
}
```